src/components/Publication/Publication.js

```diff
@@ -76,31 +76,28 @@ class Publication extends Component {
     };
 
     this.setSubdoc = this.setSubdoc.bind(this);
+    this.setArethusaLoaded = this.setArethusaLoaded.bind(this);
 
     this.arethusa = new ArethusaWrapper();
   }
 
   componentDidMount() {
-    // eslint-disable-next-line no-undef
-    this.interval = window.setInterval(this.setSubdoc, 100);
+    document.body.addEventListener('ArethusaLoaded',this.setArethusaLoaded);
   }
 
   componentWillUnmount() {
-    // eslint-disable-next-line no-undef
-    window.clearInterval(this.interval);
+    document.body.removeEventListener('ArethusaLoaded');
   }
 
+  setArethusaLoaded() {
+    document.body.removeEventListener('ArethusaLoaded');
+    this.setSubdoc();
```

Comment:
One thing we'll have to do now that we're removing the timer is to make sure that `this.setSubdoc()` runs when a user clicks `Next` or `Back`. We should be able to use [componentDidUpdate](https://reactjs.org/docs/react-component.html#componentdidupdate). We'd still need to guard against the case where Arethusa is not loaded, but we can use something like this code:

```javascript
componentDidUpdate(_prevProps, prevState) {
  const newSubDoc = this.arethusa.getSubdoc();
  const { subDoc } = prevState;

  if (subDoc !== newSubDoc) {
    this.setState({ subDoc: newSubDoc });
  }
}
```